From 0e7047c1533e5e424b28959488e8ffa91613abd9 Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jhrozek@redhat.com>
Date: Mon, 13 Feb 2017 11:46:06 +0100
Subject: [PATCH 66/79] SBUS: Document how to free the result of
 sbus_create_message
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

It might not be apparent how to free the message constructed by
sbus_create_message(). This patch just adds a comment that tells the
developer to either free the parent context or unref the message with a
dbus call directly.

Reviewed-by: Pavel Březina <pbrezina@redhat.com>
---
 src/sbus/sssd_dbus_utils.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/sbus/sssd_dbus_utils.h b/src/sbus/sssd_dbus_utils.h
index 74c21fb7930c7f5f5417b6a2587cf691b1bc0b19..e53a7faef10a4066b58213741cd73f7411419f00 100644
--- a/src/sbus/sssd_dbus_utils.h
+++ b/src/sbus/sssd_dbus_utils.h
@@ -25,6 +25,13 @@ errno_t sbus_talloc_bound_message(TALLOC_CTX *mem_ctx, DBusMessage *msg);
 errno_t sbus_error_to_errno(DBusError *error);
 errno_t sbus_check_reply(DBusMessage *reply);
 
+/* Creates a DBusMessage from a vararg list. Please note that even though
+ * this function and sbus_create_message accept a talloc memory context,
+ * it is not valid to free the resulting message with talloc_free() directly.
+ * Instead, either free the parent memory context or directly call
+ * dbus_message_unref on the message if you pass NULL memory context to
+ * these functions
+ */
 DBusMessage *sbus_create_message_valist(TALLOC_CTX *mem_ctx,
                                         const char *bus,
                                         const char *path,
-- 
2.9.3

